visualization/app/codeCharta/util/gameObjectsParser/gameObjectsImporter.ts

```diff
@@ -0,0 +1,138 @@
+import md5 from "md5"
+
+function isFile(names) {
+	return names.length === 0
+}
+
+function nodeAlreadyExists(nodes, name) {
+	return nodes.some(node => node.name === name)
+}
+
+function addNodeRecursively(names, nodes, gameObjectPosition, parentNodeName, gameObjectPositions) {
+	if (names.length === 0) return
+	// get current path name
+	const name = names.shift()
+	let node
+
+	// create node
+	node = { name, type: isFile(names) ? "File" : "Folder", attributes: {} }
+	if (!isFile(names)) node.children = []
+
+	// wrap file
+	if (isFile(names)) {
+		const childNode = { ...node }
+		childNode.attributes = { height: gameObjectPosition.scale.y }
+		node = { name, type: "Folder", attributes: {}, children: [childNode] }
+	}
+
+	// add or select node
+	if (!nodeAlreadyExists(nodes, name)) {
+		nodes.push(node)
+	} else {
+		node = nodes.find(node => node.name === name)
+	}
+
+	const parent = gameObjectPositions.find(gameObject => gameObject.name === parentNodeName)
+
+	// set Position
+	if (node.type === "Folder" && name !== "base") {
+		const cornerXofParent = parent.position.x - parent.scale.x / 2
+		const cornerZofParent = parent.position.z - parent.scale.z / 2
+		const cornerXofChild = gameObjectPosition.position.x - gameObjectPosition.scale.x / 2
+		const cornerZofChild = gameObjectPosition.position.z - gameObjectPosition.scale.z / 2
+		const top = Math.floor(((cornerXofChild - cornerXofParent) / parent.scale.x) * 100)
+		const left = Math.floor(((cornerZofChild - cornerZofParent) / parent.scale.z) * 100)
+		const width = Math.floor((gameObjectPosition.scale.z / parent.scale.z) * 100)
+		const height = Math.floor((gameObjectPosition.scale.x / parent.scale.x) * 100)
+		node.fixedPosition = { left, top, width, height }
+	}
+	if (name === "root") {
+		node.fixedPosition.top = Math.floor(50 - node.fixedPosition.height / 2)
+		node.fixedPosition.left = 0
+	}
+
+	const newParentName = parentNodeName === "base" ? node.name : `${parentNodeName}.${node.name}`
+	addNodeRecursively(names, node.children, gameObjectPosition, newParentName, gameObjectPositions)
+	return
+}
+
+function addWrappedFolderName(filePath) {
+	const splitFilePath = filePath.split(".")
+	return `/base/${filePath.replace(/\./g, "/")}/${splitFilePath.slice(-1)}`
+}
+
+function createEdge(cycle) {
+	return {
+		fromNodeName: addWrappedFolderName(cycle.from),
+		toNodeName: addWrappedFolderName(cycle.to),
+		attributes: {
+			coupling: 100
+		}
+	}
+}
+
+function createAttributeTypes() {
+	return {
+		edges: {
+			coupling: "relative"
+		}
+	}
+}
+
+export function parseGameObjectsFile(data) {
+	const { gameObjectPositions, cycles } = JSON.parse(data)
+
+	const ccJson = {
+		checksum: "",
+		data: {
+			projectName: "ScriptProjectName",
+			fileChecksum: "",
+			apiVersion: "1.3",
+			nodes: []
+		}
+	}
+
+	const nodes = []
```

Comment:
nodes never contains more than one object in this function (dummyNode). You can just create the list with that element in it.

Reply:
You are totally right!
Corrected in the next commit.